cola/main.py

```diff
@@ -0,0 +1,480 @@
+    return application_start(context, view)
+
+
+def cmd_version(args):
+    from cola import version
+    version.print_version(brief=args.brief)
+    return 0
+
+
+def cmd_pull(args):
+    context = application_init(args, update=True)
+    from cola.widgets import remote
+    view = remote.pull()
+    if args.rebase:
+        view.set_rebase(True)
+    return application_start(context, view)
+
+
+def cmd_push(args):
+    context = application_init(args, update=True)
+    from cola.widgets import remote
+    view = remote.push()
+    return application_start(context, view)
+
+
+def cmd_rebase(args):
+    kwargs = {
+            'verbose': args.verbose,
+            'quiet': args.quiet,
+            'autostash': args.autostash,
+            'fork_point': args.fork_point,
+            'onto': args.onto,
+            'preserve_merges': args.preserve_merges,
+            'strategy': args.strategy,
+            'no_ff': args.no_ff,
+            'merge': args.merge,
+            'exec': getattr(args, 'exec', None), # python keyword
+            'keep_empty': args.keep_empty,
+            'force_rebase': args.force_rebase,
+            'strategy_option': args.strategy_option,
+            'stat': args.stat,
+            'no_stat': args.no_stat,
+            'verify': args.verify,
+            'rerere_autoupdate': args.rerere_autoupdate,
+            'root': args.root,
+            'autosquash': args.autosquash,
+            'committer_date_is_author_date': args.committer_date_is_author_date,
+            'ignore_date': args.ignore_date,
+            'whitespace': args.whitespace,
+            'ignore_whitespace': args.ignore_whitespace,
+            'C': args.context_lines,
+            'continue': getattr(args, 'continue', False), # python keyword
+            'abort': args.abort,
+            'skip': args.skip,
+            'edit_todo': args.edit_todo,
+            'upstream': args.upstream,
+            'branch': args.branch,
+            'capture_output': False,
+    }
+    status, out, err = cmds.do(cmds.Rebase, **kwargs)
+    if out:
+        core.stdout(out)
+    if err:
+        core.stderr(err)
+    return status
+
+
+def cmd_remote(args):
+    context = application_init(args)
+    from cola.widgets import editremotes
+    view = editremotes.new_remote_editor()
+    return application_start(context, view)
+
+
+def cmd_search(args):
+    context = application_init(args)
+    from cola.widgets.search import search
+    view = search()
+    return application_start(context, view)
+
+
+def cmd_stash(args):
+    context = application_init(args)
+    from cola.widgets.stash import stash
+    view = stash()
+    return application_start(context, view)
+
+
+def cmd_tag(args):
+    context = application_init(args)
+    from cola.widgets.createtag import new_create_tag
+    view = new_create_tag(name=args.name, ref=args.ref, sign=args.sign)
+    return application_start(context, view)
+
+
+# Windows shortcut launch features:
```

Comment:
Would it be possible to have the `windows_find_git()` logic live in the installer so that `git-cola` itself doesn't need to contain these platform-specific details?

I was thinking that the installer could do these checks and then write bin dir it finds to `~/.config/git-cola/git-bindir` (or some other path) so that this code can simply checks for the existence of that file when running on windows and then update `$PATH` accordingly.  That way we don't have to have `C:\ProgramFiles` installation details in the code itself, if at all possible.